node/store.go

```diff
@@ -14,65 +14,75 @@ import (
 	"context"
 
 	"github.com/sourcenetwork/defradb/datastore"
-	"github.com/sourcenetwork/defradb/datastore/badger/v4"
-	"github.com/sourcenetwork/defradb/datastore/memory"
 )
 
+type StoreType string
+
+const (
+	// The Go-enum default StoreType.
+	//
+	// The actual store type that this resolves to depends on the build target.
+	DefaultStore StoreType = ""
+)
+
+// storeConstructors is a map of [StoreType]s to store constructors.
+//
+// Is is populated by the `init` functions in the runtime-specific files - this
+// allows it's population to be managed by build flags.
+var storeConstructors = map[StoreType]func(ctx context.Context, options *StoreOptions) (datastore.Rootstore, error){}
+
 // StoreOptions contains store configuration values.
 type StoreOptions struct {
-	path             string
-	inMemory         bool
-	defraStore       bool
-	valueLogFileSize int64
-	encryptionKey    []byte
+	store               StoreType
+	badgerPath          string
+	badgerFileSize      int64
+	badgerEncryptionKey []byte
+	badgerInMemory      bool
 }
 
 // DefaultStoreOptions returns new options with default values.
 func DefaultStoreOptions() *StoreOptions {
 	return &StoreOptions{
-		inMemory:         false,
-		valueLogFileSize: 1 << 30,
+		badgerInMemory: false,
+		badgerFileSize: 1 << 30,
 	}
 }
 
 // StoreOpt is a function for setting configuration values.
 type StoreOpt func(*StoreOptions)
 
-// WithInMemory sets the in memory flag.
-func WithInMemory(inMemory bool) StoreOpt {
+// WithStoreType sets the store type to use.
+func WithStoreType(store StoreType) StoreOpt {
 	return func(o *StoreOptions) {
-		o.inMemory = inMemory
+		o.store = store
 	}
 }
 
-// WithDefraStore sets the defra store flag.
-//
-// Setting this to true will result in the defra node being created with
-// the a custom defra implementation of the rootstore instead of badger.
-func WithDefraStore(defraStore bool) StoreOpt {
+// WithBadgerInMemory sets the badger in memory option.
+func WithBadgerInMemory(enable bool) StoreOpt {
```

Comment:
todo: I think if the badger store is protected by build flags, then all of the production badger stuff should also be, including these functions.  Please move them into a build-tag specific file, or explain why that is not practical :)

Reply:
The trade-off to make here is: if we move the options to the build specific file, then we will need to add platform specific files to the test harness as well. I was hoping the node package could serve as the entry point for all platforms so that we can avoid adding platform specific files elsewhere. I'm fine with either option.

Reply:
I moved the options to the badger specific file like we discussed.